### ISSUE TITLE
Add more file path matching options in Warehouse Git Subscription includePaths/excludePaths filters

### ISSUE THREAD

**Issue by maksimstankevic** [akuity/kargo#1711]
Now only regexp matching is there, enabled by "regex:/regexp:" prefix in the filter string:

```
spec:
  subscriptions:
  - git:
      branch: helm
      commitSelectionStrategy: NewestFromBranch
      includePaths:
      - regex:.*
```

 in previous discussions we talked about the need for other matching strategies to be added, like bash glob or .gitignore format.

@krancour @hiddeco I would like to start working on this and, for now, add simple-bash-glob type backed by https://pkg.go.dev/path/filepath#Match, enabled by something like a "pattern:" prefix:

```
spec:
  subscriptions:
  - git:
      branch: helm
      commitSelectionStrategy: NewestFromBranch
      includePaths:
      - pattern:charts/chart-name/*
```

wdyt?

**Comment by krancour** [comment]
I was actually thinking that bash/glob style matching is a sensible default and doesn't need to be prefixed with anything.

@hiddeco and @maksimstankevic what would you guys think about that?

**Comment by maksimstankevic** [comment]
> I was actually thinking that bash/glob style matching is a sensible default and doesn't need to be prefixed with anything.
> 
> @hiddeco and @maksimstankevic what would you guys think about that?

@krancour  I'm in! It's making the design foolproof. When thinking about this yesterday I actually felt it was cumbersome, just did not see the option of leaving the identifier out. @hiddeco wdyt?

**Comment by hiddeco** [comment]
This SGTM, thanks for picking this up @maksimstankevic 🙇

**Comment by krancour** [comment]
This was addressed by #1778
